engine/apps/alerts/models/channel_filter.py

```diff
@@ -107,20 +118,29 @@ def select_filter(cls, alert_receive_channel, raw_request_data, title, message=N
 
         satisfied_filter = None
         for _filter in filters:
-            if satisfied_filter is None and _filter.is_satisfying(raw_request_data, title, message):
+            if satisfied_filter is None and _filter.is_satisfying(raw_request_data):
                 satisfied_filter = _filter
 
         return satisfied_filter
 
-    def is_satisfying(self, raw_request_data, title, message=None):
-        return self.is_default or self.check_filter(json.dumps(raw_request_data)) or self.check_filter(str(title))
+    def is_satisfying(self, raw_request_data):
+        return self.is_default or self.check_filter(raw_request_data)
 
     def check_filter(self, value):
-        try:
-            return re.search(self.filtering_term, value)
-        except re.error:
-            logger.error(f"channel_filter={self.id} failed to parse regex={self.filtering_term}")
-            return False
+        if self.filtering_term_type == ChannelFilter.FILTERING_TERM_TYPE_JINJA2:
+            try:
+                is_matching = apply_jinja_template(self.filtering_term, payload=value)
+                return is_matching.strip().lower() in ["1", "true", "ok"]
+            except (JinjaTemplateError, JinjaTemplateWarning):
+                logger.error(f"channel_filter={self.id} failed to parse jinja2={self.filtering_term}")
+                return False
+        if self.filtering_term is not None and self.filtering_term_type == ChannelFilter.FILTERING_TERM_TYPE_REGEX:
+            try:
+                return re.search(self.filtering_term, json.dumps(value))
```

Comment:
json.dumps(value) is introduced here, could you please describe why?

Reply:
it's moved from here https://github.com/grafana/oncall/pull/1319/files#diff-5b78eae626bcf20076ff2f5f25daa3f5703ff8dd9336e5d3e3dbbd7aa827d6b2L116